File: desktop/src/at/fhooe/mc/emg/desktop/ui/dialog/VisualYMaxDialog.java
package at.fhooe.mc.emg.desktop.ui.dialog;

import javax.swing.*;
import javax.swing.border.EmptyBorder;
import javax.swing.border.TitledBorder;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class VisualYMaxDialog extends JDialog implements ActionListener {

	public interface OnVisualMaxEnteredListener {

		void onVisualMaxEntered(double max);
	}

	private static final long serialVersionUID = 1L;

	private JButton doneButton;
	
	private OnVisualMaxEnteredListener listener;
	private JTextField textMax;

	public static void show(OnVisualMaxEnteredListener listener, JFrame mainWindow) {
	    VisualYMaxDialog dialog = new VisualYMaxDialog(listener, mainWindow);
        dialog.setVisible(true);
	}

	private VisualYMaxDialog(OnVisualMaxEnteredListener listener, JFrame mainWindow) {
		setIconImage(Toolkit.getDefaultToolkit()
                .getImage(System.getProperty("user.dir") + "/desktop/icons/ic_dialog_graph_max.png"));
		this.listener = listener;
		int WIDTH = 200;
		int HEIGHT = 120;
		setBounds(mainWindow.getX() + (mainWindow.getWidth()/2) - (WIDTH /2), mainWindow.getY() + (mainWindow.getHeight()/2) - (HEIGHT /2), WIDTH, HEIGHT);
		setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
		setResizable(false);
		setTitle("Maximum visualView value");
		getContentPane().setLayout(new BorderLayout());

		JPanel contentPanel = new JPanel();
		contentPanel.setLayout(new BorderLayout());
		contentPanel.setBorder(new EmptyBorder(8, 8, 8, 8));

		doneButton = new JButton("Set visualView max");
		doneButton.addActionListener(this);
		
		getContentPane().add(contentPanel, BorderLayout.CENTER);
		
		textMax = new JTextField();
		textMax.setBorder(new TitledBorder(null, "Y-Axis maximum", TitledBorder.LEADING, TitledBorder.TOP, null, null));
		textMax.addActionListener(this);
		contentPanel.add(textMax, BorderLayout.CENTER);
		textMax.setColumns(10);
		getContentPane().add(doneButton, BorderLayout.SOUTH);
	}

	@Override
	public void actionPerformed(ActionEvent e) {

		if (e.getSource() == doneButton) {
			validateData();
		} else if (e.getSource() == textMax) {
			validateData();
		}
	}
	
	private void validateData() {
		
		double max = getValue();
		if (max > 0) {
			if (listener != null) {
				listener.onVisualMaxEntered(max);
				dispose();
			}
		} else {
			JOptionPane.showMessageDialog(this, "Please enter a number bigger than 0.0","Parse error", JOptionPane.ERROR_MESSAGE);
		}
	}
	
	private double getValue() {
		
		double val = -1;
		String text = textMax.getText();
		try {
			val = Integer.parseInt(text);
		} catch(Exception e) {
			e.printStackTrace();
		}
		return val;
	}

}

File: desktop/src/at/fhooe/mc/emg/desktop/ui/dialog/VisualYMaxDialog.kt
package at.fhooe.mc.emg.desktop.ui.dialog

import javax.swing.*
import javax.swing.border.EmptyBorder
import javax.swing.border.TitledBorder
import java.awt.*
import java.awt.event.ActionEvent
import java.awt.event.ActionListener

class VisualYMaxDialog private constructor(private val listener: OnVisualMaxEnteredListener?,
                                           mainWindow: JFrame) : JDialog(), ActionListener {

    private val doneButton: JButton
    private val textMax: JTextField

    private val value: Double
        get() {
            val text = textMax.text
            return try {
                Integer.parseInt(text).toDouble()
            } catch (e: Exception) {
                e.printStackTrace()
                (-1).toDouble()
            }
        }

    interface OnVisualMaxEnteredListener {

        fun onVisualMaxEntered(max: Double)
    }

    init {
        setIconImage(Toolkit.getDefaultToolkit()
                .getImage(System.getProperty("user.dir") + "/desktop/icons/ic_dialog_graph_max.png"))
        val width = 200
        val height = 120
        setBounds(mainWindow.x + mainWindow.width / 2 - width / 2,
                mainWindow.y + mainWindow.height / 2 - height / 2,
                width,
                height)
        defaultCloseOperation = JDialog.DISPOSE_ON_CLOSE
        isResizable = false
        title = "Maximum visualView value"
        contentPane.layout = BorderLayout()

        val contentPanel = JPanel()
        contentPanel.layout = BorderLayout()
        contentPanel.border = EmptyBorder(8, 8, 8, 8)

        doneButton = JButton("Set visualView max")
        doneButton.addActionListener(this)

        contentPane.add(contentPanel, BorderLayout.CENTER)

        textMax = JTextField()
        textMax.border = TitledBorder(null, "Y-Axis maximum", TitledBorder.LEADING, TitledBorder.TOP,
                null, null)
        textMax.addActionListener(this)
        contentPanel.add(textMax, BorderLayout.CENTER)
        textMax.columns = 10
        contentPane.add(doneButton, BorderLayout.SOUTH)
    }

    override fun actionPerformed(e: ActionEvent) {

        if (e.source === doneButton) {
            validateData()
        } else if (e.source === textMax) {
            validateData()
        }
    }

    private fun validateData() {

        val max = value
        if (max > 0) {
            if (listener != null) {
                listener.onVisualMaxEntered(max)
                dispose()
            }
        } else {
            JOptionPane.showMessageDialog(this,
                    "Please enter a number bigger than 0.0", "Parse error", JOptionPane.ERROR_MESSAGE)
        }
    }

    companion object {

        fun show(listener: OnVisualMaxEnteredListener, mainWindow: JFrame) {
            val dialog = VisualYMaxDialog(listener, mainWindow)
            dialog.isVisible = true
        }
    }

}
